connection.go

```diff
@@ -807,8 +807,8 @@ func (conn *Connection) reader(r io.Reader, c Conn) {
 			conn.reconnect(err, c)
 			return
 		}
-		resp := &Response{buf: smallBuf{b: respBytes}}
-		err = resp.decodeHeader(conn.dec)
+		resp := &ConnResponse{buf: smallBuf{b: respBytes}}
+		respHeader, err := decodeHeader(conn.dec, &resp.buf)
```

Comment:
It would be nice to swap things here:

1. Read a header from data.
2. Create a response with a header and a body data.

Reply:
Updated.